src/query/functions/src/aggregates/aggregate_string_agg.rs

```diff
@@ -44,6 +55,115 @@ pub struct StringAggState {
     values: String,
 }
 
+#[derive(BorshSerialize, BorshDeserialize, Debug)]
+struct SortDesc {
+    index: usize,
+    ty: DataType,
+    asc: bool,
+    null_first: bool,
+}
+
+#[derive(BorshSerialize, BorshDeserialize, Debug)]
+pub struct SortStringAggState {
+    inner: ArrayAggState<AnyType>,
+    sort_descs: Vec<AggregateFunctionSortDesc>,
+    delimiter: String,
+}
+
+impl Default for SortStringAggState {
+    fn default() -> Self {
+        SortStringAggState {
+            inner: Default::default(),
+            delimiter: "".to_string(),
+            sort_descs: vec![],
+        }
+    }
+}
+
+impl ScalarStateFunc<AnyType> for SortStringAggState {
+    fn new() -> Self {
+        Self::default()
+    }
+
+    fn add(&mut self, other: Option<<AnyType as ValueType>::ScalarRef<'_>>) {
+        self.inner.add(other);
+    }
+
+    fn add_batch(
+        &mut self,
+        column: &<AnyType as ValueType>::Column,
+        validity: Option<&Bitmap>,
+    ) -> Result<()> {
+        self.inner.add_batch(column, validity)
+    }
+
+    fn merge(&mut self, rhs: &Self) -> Result<()> {
+        self.inner.merge(&rhs.inner)
+    }
+
+    fn merge_result(&mut self, result_builder: &mut ColumnBuilder) -> Result<()> {
+        let values = &self.inner.values;
+        let mut builders = Vec::with_capacity(values.len());
+        builders.push(ColumnBuilder::with_capacity(
+            &Box::new(DataType::String),
+            values.len(),
+        ));
+        builders.push(ColumnBuilder::with_capacity(
+            &Box::new(DataType::String),
+            values.len(),
+        ));
+        for desc in self.sort_descs.iter() {
+            builders.push(ColumnBuilder::with_capacity(&desc.data_type, values.len()));
```

Comment:
Aggregate function will remove `NULL` value automatically, if there is a `NULL` value in the sorted column, the row will be removed, resulting in incorrect result. In this case, we need to add the function name to `NEED_NULL_AGGREGATE_FUNCTIONS` and manually remove the `NULL` value when merging the results.
https://github.com/databendlabs/databend/blob/main/src/query/functions/src/aggregates/aggregate_function_factory.rs#L30